dataprofiler/profilers/utils.py

```diff
@@ -364,11 +391,28 @@ def biased_kurt(df_series):
         return -3.0
 
     with np.errstate(all="ignore"):
-        kurt = n * M4 / np.power(M2, 2) - 3
+        kurt: float = n * M4 / np.power(M2, 2) - 3
     return kurt
 
 
-def find_diff_of_numbers(stat1, stat2):
+class Subtractable(Protocol):
+    """Protocol for annotating subtractable types."""
+
+    @abstractmethod
+    def __eq__(self: object, other: object) -> bool:
+        """Compare two objects."""
+        pass
+
+    @abstractmethod
+    def __sub__(self: T, other: T) -> Any:
+        """Compare two subtractables."""
+        pass
+
+
+T = TypeVar("T", bound=Subtractable)
+
+
+def find_diff_of_numbers(stat1: Optional[T], stat2: Optional[T]) -> Any:
```

Comment:
I think that's fine for now, but in the log run we should try to avoid using `Any` where possible.

Ideally we could create a typing file we use that makes these descriptive declarations. As things like `any` or `Optional[T]` don't make sense to someone reading it without diving deep into the code.

Will approve though, a future state for which to strive.